go/firewallsso/lib/base.go

```diff
@@ -0,0 +1,135 @@
+package libfirewallsso
+
+import (
+	"context"
+	"fmt"
+	log "github.com/inconshreveable/log15"
+	"github.com/inverse-inc/packetfence/go/logging"
+	"github.com/inverse-inc/packetfence/go/pfconfigdriver"
+	"net"
+)
+
+// Basic interface that all FirewallSSO must implement
+type FirewallSSOInt interface {
+	init(ctx context.Context) error
+	logger(ctx context.Context) log.Logger
+	Start(ctx context.Context, info map[string]string, timeout int) bool
+	Stop(ctx context.Context, info map[string]string) bool
+	GetFirewallSSO(ctx context.Context) *FirewallSSO
+	IsRoleBased(ctx context.Context) bool
```

Comment:
Having both IsRoleBased and MatchesRole seems redundant to me.

Just calling MatchesRole should be enough.